docs/pages/extensibility/ui.md

```diff
@@ -39,8 +39,94 @@ Meshery leverages remote providers for identity management. These providers can
 
 The user avatar behavior, which changes based on the user's status, can be customized by extending the [User Component](https://github.com/meshery/meshery/blob/7de49ef4928f114080f923f2ad261f4433ca91d6/ui/components/User.js#L46).
 
+### Extensibility: Permissions (CASL)
 
-###  Build-Time UI Extensibility
+Meshery UI uses [CASL.js](https://casl.js.org/v6/en/) to implement it's permissions framework. CASL is an isomorphic authorization JavaScript library which restricts what resources a given client is allowed to access.
+It has been designed to be very customizable and robust.
+It uses keys as it's unit checks to see if a permission to perform or action or not.
```

Comment:
"unit checks"?

Reply:
Include one statement about CASL and that's it. This isn't about CASL. This is about Meshery's extensible permission framework, how it works, and how to use it to extend Meshery. How are new keys added? Do they ever change? Where is the current list? Is there an example extension? Why use these?